:bug: Fixed AssetHelper not working with svg

torstenzander opened the pull request:
closes #13971

This fixes an issue with svg sprites using the # to select an icon, when a ?v=hash is rendered the svg images is not shown. The bugfix removes ?v=hash param for svg files with an #.

Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

More info can be found by clicking the "guidelines for contributing" link above.

torstenzander committed:
:bug: Fixed AssetHelper not working with svg
closes #13971

This fixes an issue with svg sprites using the # to select an icon, when a ?v=hash is rendered the svg images is not shown. The bugfix removes ?v=hash param for svg files with an #.

ErisDS commented:
Hey @torstenzander thanks for the PR :) 

I don't think this is the fix that you want. The `?v=` query param is for cache busting, so this change would mean any svg with a # would get stuck in cache indefinitely.

The correct fix would be to insert the `?v=` param before the #, and maintain both.

Also, this bug would affect absolutely any path with a #, although it currently only really makes sense to use with an SVG, there's no need to limit the solution if implemented this way  🙂

ErisDS commented on test/unit/frontend/meta/asset-url.test.js:
```diff
@@ -34,6 +34,11 @@ describe('getAssetUrl', function () {
 
+    it('should return no hash when svg#', function () {
+        const testUrl = getAssetUrl('myfile.svg#arrow-up');
+        testUrl.should.equal('/assets/myfile.svg#arrow-up');
```
```
testUrl.should.equal(`/assets/myfile.svg?v=${config.get('assetHash')}#arrow-up`);
```

I'd expect the test to look like this :)

ErisDS reviewed (changes requested):
Hey @torstenzander thanks for the PR :)

I don't think this is the fix that you want. The ?v= query param is for cache busting, so this change would mean any svg with a # would get stuck in cache indefinitely.

The correct fix would be to insert the ?v= param before the #, and maintain both.

Also, this bug would affect absolutely any path with a #, although it currently only really makes sense to use with an SVG, there's no need to limit the solution if implemented this way 🙂

torstenzander committed:
:bug: Fixed AssetHelper not working with svg
closes #13971

This fixes an issue with links containing # anchor. It makes sure the # part is at the end of the url like url?v=hash#anhor

torstenzander commented:
I am not sure if I have to do anything, my tests cover everything I added even though the coverage report looks different.

ErisDS committed:
Merge branch 'main' into main

ErisDS reviewed (approved):
Hey @torstenzander no you don't need to do anything. Thanks for your work :)
